README.md

```diff
@@ -130,343 +135,328 @@ ag https://raw.githubusercontent.com/asyncapi/generator/v1.0.1/test/docs/dummy.y
 
-##### Official AsyncAPI Filters
+### Using JS in template
 
-AsyncAPI Initiative provides a library of filters. You can also create such a library for your templates. You add such a library to `dependencies` in the `package.json` file and configure in the `generator.filters` section like 
-```json
-{
-  ...
-  "generator": {
-    ...
-    "filters": [
-      "@asyncapi/generator-filters"
-    ]
-  }
+When you use React, you are actually using JS, so you can apply conditions to rendering, split functionality into separate/reusable functions, create compisition, extend/mix functions etc.
+
+```js
+/* 
+ * If asyncapi has `externalDocs` property then the Generator will return appropriate string,
+ * otherwise it won't render anything.
+ */
+function ExternalDocs({ asyncapi }) {
+  if (!asyncapi.hasExternalDocs()) return null;
+  return `Don't forget to visit our website ${asyncapi.externalDocs().url()}.`
 }
 ```
 
-What you get in exchange:
-- Access to all the filters listed [here](https://github.com/asyncapi/generator-filters/blob/master/docs/api.md#functions) like super useful `log` filter that works similar to Nunjucks `dump` filter but instead of dumping information in the template file, it logs it to the terminal.
-- Access to all [Lodash](https://lodash.com/docs/4.17.15) functions that are made available as filters
+### Retrieve rendered content from children
 
-Check [partials/diagramContent.html](partials/diagramContent.html) to see how `log` filter is used. As you can see, it works like any other Nunjucks filter
-```html
-{{ "Template cannot generate proper diagram because you have no schemas defined in components section" | log }}
-```
+Each component has a `childrenContent` property. It is the processed children content of a component into a pure string. You can use it for compositions in your component.
+
+```js
+function CustomComponent({ childrenContent }) {
+  return `some text at the beginning: ${childrenContent}`
+}
 
-Also have a look at [template/schemas/$$schema$$-example.html](template/schemas/$$schema$$-example.html) file to see `generateExample` filter in action:
-```html
-<pre class="hljs mb-4 border border-grey-darkest rounded">
-    <code>
-        <!-- Here you can see an example of chaining filters and also usage of a filter provided by @asyncapi/generator-filters package -->
-        {{ schema.json() | generateExample | safe }}
-    </code>
-</pre>
+function RootComponent() {
+  return (
+    <CustomComponent>
+      some text at the end.
+    </CustomComponent>
+  );
+}
 ```
 
-##### Template Custom Filters
+Then output from `RootComponent` will be `some text at the beginning: some text at the end.`.
 
-Template custom filters are the ones attached to your Template. Check [Template specific filters](#template-specific-filters) section to understand how they work.
+### Render component to string
```

Comment:
this section and 3more before should be `####` so they end up under `Template` section in toc